prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -283,6 +291,24 @@ class ReportFunction(
         return responseBuilder.build()
     }
 
+    fun responseContainsQualityFilter(response: String): Boolean {
+        var hasQualityFilters = false
+        val tree = jacksonObjectMapper().readTree(response)
```

Comment:
The response from the client may not be valid json.

I think you might need some error handling here? Does invalid json throw?

Reply:
@TomNUSDS The response this method is evaluating is from us, our own endpoint (as opposed to the /settings endpoint payload, which has a much greater chance of receiving bad json). So if that json is invalid, we have a lot more problems than just here. Also, I did add a couple unit tests around this method